src/Modules/ProjectModule/components/ProjectData/ProjectData.tsx

```diff
@@ -1,6 +1,121 @@
+import { Link, useLocation } from "react-router-dom";
+import "./ProjectData.css";
+import { useForm } from "react-hook-form";
+import { AuthContext } from "../../../../context/AuthContext";
+import { useContext } from "react";
+import axios from "axios";
+import { toast } from "react-toastify";
 
 export default function ProjectData() {
+  interface projectData{
+    title:string,
+    description:string,
+    manager?:{},
+    id:number,
+    creationDate:string,
+    modificationDate:string
+
+  }
+  let location = useLocation()
+  console.log(location);
+  let status = location.state?.type === 'update';
+  let updateData = location.state?.updateData
+ 
+ 
+  
+  let {baseUrl} = useContext(AuthContext)
+  let {register,handleSubmit,formState:{errors},reset}=useForm <projectData>();
+
+  const onsubmitForm = async (data:projectData)=>{
+   try {
+    let respons = await axios({
+      method: status ?'put':'post',
+      url:status?`${baseUrl}/Project/${updateData.id}`: `${baseUrl}/Project`,
+      data:data,
+      headers:{ Authorization: `Bearer ${localStorage.getItem("token")}` }
+    })
+    status? toast.success('your Project Is Updated '):toast.success('your Project Is Created')
```

Comment:
We can make the condition inside  the success function instead of repeating toast.success two times